Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -56,7 +56,7 @@ struct NavigationOptions {
   const object_t* endObject = nullptr;
 
   /// External surface identifier for which the boundary check is ignored
-  std::vector<GeometryIdentifier> externalSurfaces = {};
+  std::vector<const Surface*> externalSurfaces{};
```

Comment:
one thing we need to consider is if this should take ownership of the surfaces or if raw pointers are fine. if these surfaces are not part of the geometry, the user would have to take care of their lifetime right now